src/PSRule.Rules.Azure/rules/Azure.VM.Rule.ps1

```diff
@@ -247,3 +247,26 @@ Rule 'Azure.VM.PPGName' -Ref 'AZR-000260' -Type 'Microsoft.Compute/proximityPlac
 }
 
 #endregion Proximity Placement Groups
+
+# Synopsis: Protect Custom Script Extensions commands
+Rule 'Azure.VM.ScriptExtensions' -Ref 'AZR-000290' -Type 'Microsoft.Compute/virtualMachines', 'Microsoft.Computer/virtualMachines/CustomScriptExtension', 'Microsoft.Compute/virtualMachines/extensions' -Tag @{ release = 'GA'; ruleSet = '2022_09' } {
+    $vmConfig = @($TargetObject);
+
+    if ($PSRule.TargetType -eq 'Microsoft.Compute/virtualMachines') {
+        $vmConfig = @(GetSubResources -ResourceType 'extensions' );
```

Comment:
Try both `extenstions` and `Microsoft.Compute/virtualMachines/extensions` since both are valid.